PR: https://github.com/odoo/odoo/pull/145178

From: ab3c81b680a1019ff33aabf87cf533af7bfe8cfa
From: Florian(flg)
Date: 2023-12-07 19:07:46

Structural Changes: 3
Total Changes: 17

[REM] account: Remove Customer Payments Follow up feature for sale journal

The aim of this commit is removing all the nearly dead code
that handle the Customer Payments follow up feature on
sale journal. This has been added when the follow up feature
was not mature enough. As no one is using this feature, we
decide to remove it.

closes odoo/odoo#145178

Task-id: 3627753
Related: odoo/upgrade#5416
Signed-off-by: Brice Bartoletti (bib) <bib@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account/models/account_journal.py
+++ b/addons/account/models/account_journal.py
@@ -179,11 +179,6 @@ class AccountJournal(models.Model):
     bank_acc_number = fields.Char(related='bank_account_id.acc_number', readonly=False)
     bank_id = fields.Many2one('res.bank', related='bank_account_id.bank_id', readonly=False)
 
-    # Sale journals fields
-    sale_activity_type_id = fields.Many2one('mail.activity.type', string='Schedule Activity', default=False, help="Activity will be automatically scheduled on payment due date, improving collection process.")
-    sale_activity_user_id = fields.Many2one('res.users', string="Activity User", help="Leave empty to assign the Salesperson of the invoice.")
-    sale_activity_note = fields.Text('Activity Summary')
-
     # alias configuration for journals
     alias_id = fields.Many2one(help="Send one separate email for each invoice.\n\n"
                                     "Any file extension will be accepted.\n\n"

--- a/addons/account/models/account_move.py
+++ b/addons/account/models/account_move.py
@@ -3772,18 +3772,6 @@ class AccountMove(models.Model):
                 if p not in invoice.sudo().message_partner_ids
             ])
 
-            if (
-                invoice.is_sale_document()
-                and invoice.journal_id.sale_activity_type_id
-                and (invoice.journal_id.sale_activity_user_id or invoice.invoice_user_id).id not in (self.env.ref('base.user_root').id, False)
-            ):
-                invoice.activity_schedule(
-                    date_deadline=min((date for date in invoice.line_ids.mapped('date_maturity') if date), default=invoice.date),
-                    activity_type_id=invoice.journal_id.sale_activity_type_id.id,
-                    summary=invoice.journal_id.sale_activity_note,
-                    user_id=invoice.journal_id.sale_activity_user_id.id or invoice.invoice_user_id.id,
-                )
-
         customer_count, supplier_count = defaultdict(int), defaultdict(int)
         for invoice in to_post:
             if invoice.is_sale_document():
